Input.cpp

```diff
@@ -0,0 +1,58 @@
+#pragma once
+#include "Engine.h"
+using namespace sf;
+
+
+void Engine::input()
+{
+	Event event;
+	updateMousePosition();
+
+	//Handle the player quitting
+	if (Keyboard::isKeyPressed(Keyboard::Escape))
+	{
+		m_Window.close();
+
+	}
+
+	////Handle the player starting the game
+	//if (Keyboard::isKeyPressed(Keyboard::Return))
+	//{m_Playing = true;} else {m_Playing = false;}
+
+	//Handle Mouse clicking
+	if (Mouse::isButtonPressed(Mouse::Left) && m_Playing == true)
+	{
+		
+		std::vector<sf::Sprite> &cupcakes = cupcake.m_spriteVector;

```

Comment:
So this is why it is a public member!

I want you to think of a way to remove the need for this member.

Can you make some functions that would tell you for example `int Cupcake::CountOutOfScreenAndDelete()`, then feed that result into `player.decreaseHealth(count)`.

And there could be a method `bool Cupcake::ClickOnACupcake(m_mousePosition)` that returns true and deletes the cupcake when it is clicked!